### DISCUSSION TITLE
Add Ability to add a Custom Time Range to Analytics Dashboard

### DISCUSSION THREAD

**Discussion by bradstondevcode** [forem/forem#15455] [Feature Requests]
I would love to have the ability to view a custom date range on the "Analytics Dashboard".

Currently the "Week" and "Month" are helpful but they are very temporal so if you miss the time window to view that data it may be hard to drill down and further.

Also, the "Infinity" option is not very useful as it includes dates prior to the user joining the site so it has tons of unhelpful data and it makes the data that is important near unreadable.

Is it possible to add a "Custom Range" Option where you can add a range that is relevant to you as a user.

**Comment by JennieOcken** [comment]
Really appreciate this feedback @bradstondevcode .  We are currently thinking through the dashboard holistically to improve some of those things.  I'm not sure yet where we are going to end up but I am going to cc: @ludwiczakpawel to be aware of this discussion and put this in the "up next" area as we are committed to working on dashboard soon.

**Comment by Balastrong** [comment]
Hello everyone, I was going to open a similar conversation but I see there's already this one, so I'll add my bits here.

I also think Infinity is kind of pointless unless you joined in early 2020. Having two custom calendars to pick from and to would be indeed the best option.

If there is some technical limitation (e.g. getting data from backend) a first step could be by replacing **Infinity** with something like **Beginning**, so that the chart is meaningful for everyone.
This could be even performed frontend once all data is read (just tell chart.js to start from a different date, using the same dataset).
